Frameworks/Foundation/NSKVOSupport.mm

```diff
@@ -695,15 +687,14 @@ - (void)didChange:(NSKeyValueChange)changeKind valuesAtIndexes:(NSIndexSet*)inde
                                         notifyingObject = self ](_NSKVOKeyObserver * keyObserver, NSMutableDictionary * keypathValueCache) {
             _NSKVOKeypathObserver* keypathObserver = keyObserver.keypathObserver;
             NSKeyValueObservingOptions options = keypathObserver.options;
-            id rootObject = keypathObserver.object;
-            NSString* keypath = keypathObserver.keypath;
             NSMutableDictionary* change = keypathObserver.pendingChange;
-
             if ((options & NSKeyValueObservingOptionNew) && [change[NSKeyValueChangeKindKey] integerValue] != NSKeyValueChangeRemoval) {
                 // For to-many mutations, we can't get the new values at indexes that have been deleted.
+                id rootObject = keypathObserver.object;
+                NSString* keypath = keypathObserver.keypath;
                 id newValue = keypathValueCache[keypath];
                 if (!newValue) {
-                    newValue = _valueForPendingChangeAtIndexes(notifyingObject, key, rootObject, keypath, keyObserver, indexes);
+                    newValue = _valueForPendingChangeAtIndexes(notifyingObject, key, rootObject, keyObserver, indexes);
```

Comment:
I believe this is subject to the same caveats as above.

I swear we had tests testing this.